src/models/document/document-utils.ts

```diff
@@ -15,3 +19,22 @@ export function getDocumentDisplayTitle(
             ? `${problem.title}: Planning`
             : document.getDisplayTitle(appConfig);
 }
+
+/**
+ * Returns the key for user documents or path for problem documents
+ * @param document 
+ * @returns 
+ */
+export function getDocumentIdentifier(document?: DocumentContentModelType) {
```

Comment:
Ironically enough, I was just starting to make this same change locally to reduce the `ContentModel`'s dependencies.